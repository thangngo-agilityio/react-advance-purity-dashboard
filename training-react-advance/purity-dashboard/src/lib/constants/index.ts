export * from './navbar';
export * from './footer';
export * from './router';
export * from './images';
export * from './api';
export * from './local-storage';
export * from './columns';
export * from './author';
export * from './project';
