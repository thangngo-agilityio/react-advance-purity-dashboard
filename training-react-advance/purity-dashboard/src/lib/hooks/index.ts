export * from './useAuth';
export * from './useAuthor';
export * from './useProject';
