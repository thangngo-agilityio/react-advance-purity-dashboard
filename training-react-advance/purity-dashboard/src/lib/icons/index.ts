export * from './build';
export * from './card';
export * from './chart';
export * from './document';
export * from './home';
export * from './help';
export * from './logo';
export * from './notification';
export * from './person';
export * from './rocket';
export * from './setting';
export * from './search';
export * from './line';
