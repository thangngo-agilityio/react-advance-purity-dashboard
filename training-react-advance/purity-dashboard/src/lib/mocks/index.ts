export * from './sidebar';
export * from './project';
