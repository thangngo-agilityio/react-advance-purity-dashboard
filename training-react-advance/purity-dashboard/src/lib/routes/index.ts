import { createBrowserRouter } from 'react-router-dom';
import { publicRoutes } from './public';

export const routes = createBrowserRouter([publicRoutes]);
