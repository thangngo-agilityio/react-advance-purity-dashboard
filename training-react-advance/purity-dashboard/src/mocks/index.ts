export * from './sidebar';
export * from './project';
export * from './author';
export * from './user';
