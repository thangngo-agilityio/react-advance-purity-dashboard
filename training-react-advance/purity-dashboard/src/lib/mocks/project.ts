import { AxiosRequestHeaders } from 'axios';

// Constants
import { PROJECT_STATUS } from '../constants';

// Types
import { TRecordProject } from '../types';

export const PROJECT: TRecordProject[] = [
  {
    id: 'rec393ALcB4az1FUZ',
    createdTime: '2024-08-11T17:08:32.000Z',
    fields: {
      completion: 12,
      projectName: 'Project ',
      image: 'https://d38b044pevnwc9.cloudfront.net/cutout-nuxt/enhancer/3.jpg',
      avatar:
        'https://d38b044pevnwc9.cloudfront.net/cutout-nuxt/enhancer/3.jpg',
      _id: '7',
      status: PROJECT_STATUS.WORKING,
      budget: 123213,
      description: 'asdasd',
    },
  },
];

export const PROJECT_DATA_MOCK: TRecordProject = {
  id: 'rec393ALcB4az1FUZ',
  createdTime: '2024-08-11T17:08:32.000Z',
  fields: {
    completion: 12,
    projectName: 'Project ',
    image: 'https://d38b044pevnwc9.cloudfront.net/cutout-nuxt/enhancer/3.jpg',
    avatar: 'https://d38b044pevnwc9.cloudfront.net/cutout-nuxt/enhancer/3.jpg',
    _id: '7',
    status: PROJECT_STATUS.WORKING,
    budget: 123213,
    description: 'asdasd',
  },
};

export const MOCK_ADD_PROJECT_PAYLOAD = {
  records: [
    {
      fields: {
        projectName: 'Chakra Soft UI Version',
        avatar: 'https://nhatnghe.net/uploads/news/ms.jpg',
        budget: 14000,
        status: 'Working',
        completion: 60,
        description:
          'As Uber works through a huge amount of internal management turmoil.',
      },
    },
  ],
};

export const MOCK_UPDATE_PROJECT_PAYLOAD = {
  name: 'Update name',
  productId: PROJECT[0].id,
};

export const MOCK_PROJECTS_SUCCESS_RES = {
  data: { records: PROJECT },
  status: 200,
  statusText: 'Ok',
  headers: {},
  config: {
    headers: {} as AxiosRequestHeaders,
  },
};
