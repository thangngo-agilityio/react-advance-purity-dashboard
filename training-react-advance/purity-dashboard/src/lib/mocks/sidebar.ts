import { HomeIcon, ChartIcon, CardIcon, BuildIcon } from '@/ui/icons';
import { ROUTES } from '../constants';

export const SIDEBAR_LIST_MOCK = [
  {
    id: 1,
    leftIcon: HomeIcon,
    menuItemContent: 'Dashboards',
    destination: '/',
  },
  {
    id: 2,
    leftIcon: ChartIcon,
    menuItemContent: 'Tables',
    destination: `/`,
  },
  {
    id: 3,
    leftIcon: CardIcon,
    menuItemContent: 'Billing',
    destination: `/${ROUTES.BILLING}`,
  },
  {
    id: 4,
    leftIcon: BuildIcon,
    menuItemContent: 'RTL',
    destination: `/${ROUTES.RTL}`,
  },
];
