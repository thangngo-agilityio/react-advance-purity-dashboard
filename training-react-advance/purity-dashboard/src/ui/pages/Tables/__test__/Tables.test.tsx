import { fireEvent, render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
// import dayjs from "dayjs";

// Components
import TablePage from "..";

// Hooks
// import { useAuthor } from "@/lib/hooks";

// Mocks
import { AUTHORS, PROJECT } from "@/lib/mocks";

// Constants
// import { SUCCESS_MESSAGE } from "@/lib/constants/message";

// Types
// import { TRecordAuthor } from "@/lib/types";

const mockCreateAuthor = jest.fn()
const mockUpdateAuthor = jest.fn()
const mockUpdateProject = jest.fn()

jest.mock('@/lib/hooks', () => ({
  useAuthor: () => ({
    authorData: AUTHORS,
    isLoading: false,
    isFetching: false,
    createAuthor: mockCreateAuthor,
    updateAuthor: mockUpdateAuthor
  }),
  useProject: () => ({
    projectData: PROJECT,
    isLoading: false,
    isFetching: false,
    updateProject: mockUpdateProject,
  }),
  useDebounce: jest.fn(),

}));

jest.mock('@chakra-ui/react', () => ({
  useToast: jest.fn(),
}));

const mockToast = jest.fn();

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useToast: () => mockToast,
}));


describe('Tables page', () => {

  beforeEach(() => {
    jest.clearAllMocks();

  })
  test('Should render match with snapshot.', () => {
    const { container } = render(<TablePage />);

    expect(container).toMatchSnapshot();
  });

  test('should open add modal when clicking add button', () => {
    render(<TablePage />);

    // Find the button that opens the modal
    const addButton = screen.getByText(/Add New/i);

    // Ensure the modal is not open initially
    expect(screen.queryByText(/Add Author/i)).not.toBeInTheDocument();

    // Click the button to open the modal
    fireEvent.click(addButton);

    // Check if the modal is now open
    expect(screen.getByText(/Add Author/i)).toBeInTheDocument();
  });

  // it('should create an author successfully and show a success toast', async () => {
  //   const mockData: TRecordAuthor[] = [{
  //     id: '1',
  //     createdTime: '2024-08-14',
  //     fields: {
  //       _id: 1,
  //       name: 'John Doe',
  //       email: 'john.doe@example.com',
  //       avatar: 'avatar_url',
  //       role: 'Admin',
  //       job: 'Software Engineer',
  //       employed: '2024-08-14',
  //       status: undefined, // Assuming this is optional
  //     },
  //   }];

  //   (dayjs as unknown as jest.Mock).mockReturnValue({
  //     format: jest.fn().mockReturnValue('2024-08-14'),
  //   });

  //   (mockCreateAuthor as jest.Mock).mockResolvedValue({}); // Mock the API call to resolve

  //   const { result } = renderHook(() => useAuthor());

  //   await act(async () => {
  //     await result.current.createAuthor(mockData);
  //   });

  //   expect(mockCreateAuthor).toHaveBeenCalledWith({ "records": [{ "fields": { "avatar": "avatar_url", "email": "john.doe@example.com", "employed": "2024-08-14", "job": "Software Engineer", "name": "John Doe", "role": "Admin" } }] });

  //   expect(mockToast).toHaveBeenCalledWith({
  //     title: SUCCESS_MESSAGE.TITLE_MESSAGE_CREATE('Author'),
  //     description: SUCCESS_MESSAGE.AUTHOR_SUCCESS,
  //     status: 'success',
  //     duration: 2000,
  //     isClosable: true,
  //   });
  // });
})
