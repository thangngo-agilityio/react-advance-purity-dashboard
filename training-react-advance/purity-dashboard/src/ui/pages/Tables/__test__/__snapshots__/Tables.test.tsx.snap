// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tables page Should render match with snapshot. 1`] = `
<div>
  <div
    class="chakra-stack css-1ey9w6j"
  >
    <div
      class="css-10preug"
    >
      <div
        class="css-dv5ta8"
      >
        <div
          class="css-0"
        >
          <div
            class="css-ofhpoc"
          >
            <p
              class="chakra-text css-0"
            >
              Pages /
            </p>
            <a
              class="chakra-link css-0"
              href="/tables"
            >
              Tables
            </a>
          </div>
          <h2
            class="chakra-heading css-0"
          >
            Tables
          </h2>
        </div>
      </div>
      <div
        class="css-mj64b3"
      >
        <div
          class="css-x0sdsu"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <div
                class="chakra-input__left-element css-1u2k0bl"
              >
                <svg
                  fill="none"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14.1292 13.0958L11.3726 10.3393C12.0363 9.4558 12.3945 8.38041 12.3933 7.27542C12.3933 4.45297 10.097 2.15668 7.27457 2.15668C4.45211 2.15668 2.15582 4.45297 2.15582 7.27542C2.15582 10.0979 4.45211 12.3942 7.27457 12.3942C8.37955 12.3954 9.45495 12.0371 10.3384 11.3735L13.095 14.13C13.2345 14.2547 13.4165 14.3213 13.6036 14.3161C13.7907 14.3109 13.9687 14.2342 14.101 14.1019C14.2333 13.9695 14.31 13.7915 14.3152 13.6044C14.3205 13.4174 14.2539 13.2354 14.1292 13.0958ZM3.61832 7.27542C3.61832 6.55229 3.83276 5.84539 4.23451 5.24412C4.63626 4.64286 5.20729 4.17422 5.87538 3.89749C6.54348 3.62076 7.27863 3.54835 7.98787 3.68943C8.69711 3.83051 9.34859 4.17873 9.85993 4.69007C10.3713 5.2014 10.7195 5.85288 10.8606 6.56213C11.0016 7.27137 10.9292 8.00652 10.6525 8.67461C10.3758 9.3427 9.90714 9.91373 9.30587 10.3155C8.70461 10.7172 7.99771 10.9317 7.27457 10.9317C6.30523 10.9305 5.37592 10.5449 4.6905 9.8595C4.00507 9.17407 3.61949 8.24477 3.61832 7.27542Z"
                    fill="#2D3748"
                  />
                </svg>
              </div>
              <input
                class="chakra-input css-gh4tfk"
                id="field-:r0:"
                placeholder="Type here..."
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="css-0"
        >
          <div
            class="css-2efhy9"
          >
            <svg
              fill="none"
              height="13"
              viewBox="0 0 13 13"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.50003 4.62512C6.12919 4.62512 5.76668 4.73509 5.45834 4.94111C5.14999 5.14714 4.90967 5.43998 4.76776 5.78259C4.62584 6.1252 4.58871 6.5022 4.66106 6.86591C4.73341 7.22963 4.91198 7.56372 5.17421 7.82594C5.43643 8.08817 5.77052 8.26674 6.13424 8.33909C6.49795 8.41144 6.87495 8.37431 7.21756 8.23239C7.56017 8.09048 7.85301 7.85016 8.05904 7.54181C8.26506 7.23347 8.37503 6.87096 8.37503 6.50012C8.37355 6.0033 8.17552 5.52724 7.82422 5.17593C7.47291 4.82462 6.99685 4.6266 6.50003 4.62512ZM10.5482 6.50012C10.5472 6.67532 10.5343 6.85025 10.5097 7.02371L11.6509 7.91715C11.7006 7.95819 11.7341 8.01557 11.7454 8.07902C11.7567 8.14248 11.745 8.20789 11.7125 8.26356L10.633 10.1273C10.6002 10.1825 10.549 10.2243 10.4883 10.2454C10.4277 10.2666 10.3616 10.2657 10.3016 10.2429L8.95979 9.70379C8.68039 9.91894 8.37463 10.0975 8.04995 10.2351L7.84933 11.6596C7.8381 11.7235 7.80506 11.7814 7.75586 11.8235C7.70665 11.8657 7.64434 11.8894 7.57956 11.8907H5.4205C5.3569 11.8895 5.2956 11.8667 5.24661 11.8262C5.19761 11.7856 5.16382 11.7296 5.15073 11.6674L4.95011 10.2429C4.62453 10.1068 4.31855 9.92799 4.04027 9.71106L2.69847 10.2501C2.63849 10.2729 2.57238 10.2739 2.51177 10.2528C2.45115 10.2317 2.39992 10.1899 2.36706 10.1348L1.28753 8.27129C1.25504 8.21562 1.2434 8.15022 1.25469 8.08676C1.26598 8.0233 1.29947 7.96592 1.34917 7.92488L2.49035 7.03145C2.466 6.85536 2.45317 6.67788 2.45191 6.50012C2.45288 6.32492 2.46572 6.14999 2.49035 5.97653L1.34917 5.08309C1.29947 5.04205 1.26598 4.98467 1.25469 4.92121C1.2434 4.85776 1.25504 4.79235 1.28753 4.73668L2.36706 2.87293C2.39988 2.81776 2.4511 2.77593 2.51172 2.75479C2.57234 2.73366 2.63846 2.73458 2.69847 2.75739L4.04027 3.29645C4.31967 3.0813 4.62543 2.90274 4.95011 2.76512L5.15073 1.34059C5.16196 1.27678 5.195 1.21885 5.24421 1.1767C5.29341 1.13455 5.35572 1.1108 5.4205 1.1095H7.57956C7.64316 1.11073 7.70446 1.1335 7.75345 1.17406C7.80245 1.21463 7.83625 1.27061 7.84933 1.33286L8.04995 2.75739C8.37594 2.89332 8.68232 3.07216 8.96097 3.28918L10.3016 2.75012C10.3616 2.72729 10.4277 2.72634 10.4883 2.74743C10.5489 2.76852 10.6001 2.8103 10.633 2.86543L11.7125 4.72918C11.745 4.78485 11.7567 4.85026 11.7454 4.91371C11.7341 4.97717 11.7006 5.03455 11.6509 5.07559L10.5097 5.96903C10.534 6.14503 10.5469 6.32244 10.5482 6.50012Z"
                fill="#718096"
              />
            </svg>
            <svg
              fill="none"
              height="13"
              viewBox="0 0 13 13"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.8143 8.4992C10.7754 8.45233 10.7372 8.40545 10.6997 8.36022C10.1841 7.73655 9.87213 7.36014 9.87213 5.5946C9.87213 4.68053 9.65346 3.93054 9.22244 3.36804C8.90463 2.95249 8.47502 2.63725 7.90877 2.40429C7.90148 2.40023 7.89498 2.39492 7.88955 2.38858C7.68588 1.70655 7.12854 1.24976 6.49994 1.24976C5.87135 1.24976 5.31424 1.70655 5.11057 2.38788C5.10513 2.39398 5.09872 2.39913 5.09159 2.40311C3.77018 2.9471 3.12799 3.99077 3.12799 5.59389C3.12799 7.36014 2.81651 7.73655 2.30042 8.35952C2.26292 8.40475 2.22471 8.45069 2.18581 8.4985C2.08531 8.6197 2.02163 8.76716 2.00232 8.92342C1.983 9.07967 2.00886 9.23819 2.07682 9.38022C2.22143 9.68491 2.52963 9.87405 2.88143 9.87405H10.121C10.4712 9.87405 10.7773 9.68514 10.9224 9.38186C10.9906 9.2398 11.0167 9.08115 10.9976 8.92471C10.9784 8.76827 10.9148 8.6206 10.8143 8.4992Z"
                fill="#718096"
              />
              <path
                d="M6.49996 11.7499C6.83864 11.7497 7.17093 11.6577 7.46159 11.4839C7.75225 11.31 7.99044 11.0608 8.15089 10.7625C8.15845 10.7482 8.16219 10.7322 8.16173 10.7161C8.16128 10.6999 8.15665 10.6841 8.1483 10.6703C8.13995 10.6564 8.12816 10.645 8.11408 10.637C8.09999 10.6291 8.08409 10.6249 8.06792 10.6249H4.93246C4.91627 10.6249 4.90034 10.629 4.88622 10.6369C4.87211 10.6449 4.86029 10.6563 4.85191 10.6702C4.84353 10.684 4.83888 10.6998 4.83842 10.716C4.83795 10.7322 4.84169 10.7482 4.84926 10.7625C5.00969 11.0607 5.24784 11.31 5.53846 11.4838C5.82907 11.6577 6.16132 11.7496 6.49996 11.7499Z"
                fill="#718096"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-1365pfc"
    >
      <div
        class="chakra-stack css-1qzyrcn"
      >
        <div
          class="css-1ufrzn0"
        >
          <h2
            class="chakra-heading css-1fhgjcy"
          >
            Authors Table
          </h2>
          <button
            class="chakra-button css-mkfe5m"
            type="button"
          >
            Add new 
            <svg
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="chakra-table__container css-1c13jy5"
        >
          <table
            class="chakra-table css-0"
          >
            <thead
              class="css-0"
            >
              <tr
                class="css-0"
              >
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="AUTHOR"
                    >
                      AUTHOR
                    </p>
                  </div>
                </th>
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="FUNCTION"
                    >
                      FUNCTION
                    </p>
                  </div>
                </th>
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="STATUS"
                    >
                      STATUS
                    </p>
                  </div>
                </th>
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="EMPLOYED"
                    >
                      EMPLOYED
                    </p>
                  </div>
                </th>
                <th
                  class="css-bsiblf"
                />
              </tr>
            </thead>
            <tbody
              class="css-0"
            >
              <tr
                class="css-0"
                data-testid="table-row"
              >
                <td
                  class="css-1ydvgvr"
                >
                  <div
                    class="css-fktblz"
                  >
                    <div
                      class="css-xax5g4"
                    >
                      <img
                        alt="Thang"
                        class="chakra-image css-15sm3g6"
                        src="https://tiki.vn/blog/wp-content/uploads/2023/02/o0XKdJdOcSuXoo3NbsjzNdn00-7PcHzNpAuQ85LIrsC2qwi4oEykPVZGDPskjmOQvd85etOVIPb4f0rprNu_dGGbslg_L4kUHgra7v..."
                      />
                    </div>
                    <div
                      class="chakra-stack css-1c1uyy8"
                    >
                      <h2
                        class="chakra-heading css-1ouru2w"
                      >
                        Thang
                      </h2>
                      <p
                        class="chakra-text css-1vox59v"
                      >
                        asd@gmail.com
                      </p>
                    </div>
                  </div>
                </td>
                <td
                  class="css-1nygxpw"
                >
                  <div
                    class="chakra-stack css-qa66p0"
                  >
                    <h2
                      class="chakra-heading css-fcqnrt"
                    >
                      asd
                    </h2>
                    <p
                      class="chakra-text css-fcqnrt"
                    >
                      asd
                    </p>
                  </div>
                </td>
                <td
                  class="css-10ngfz9"
                >
                  <span
                    class="chakra-badge css-11k7pgp"
                  >
                    Offline
                  </span>
                </td>
                <td
                  class="css-svc9xl"
                >
                  <p
                    class="chakra-text css-1n7ogn7"
                  >
                    01/08/24
                  </p>
                </td>
                <td
                  class="css-lepxbj"
                >
                  <button
                    class="chakra-button css-yx91ij"
                    type="button"
                  >
                    Edit
                  </button>
                </td>
              </tr>
              <tr
                class="css-0"
                data-testid="table-row"
              >
                <td
                  class="css-1ydvgvr"
                >
                  <div
                    class="css-fktblz"
                  >
                    <div
                      class="css-xax5g4"
                    >
                      <img
                        alt="asdasd"
                        class="chakra-image css-15sm3g6"
                        src="https://tiki.vn/blog/wp-content/uploads/2023/02/o0XKdJdOcSuXoo3NbsjzNdn00-7PcHzNpAuQ85LIrsC2qwi4oEykPVZGDPskjmOQvd85etOVIPb4f0rprNu_dGGbslg_L4kUHgra7v..."
                      />
                    </div>
                    <div
                      class="chakra-stack css-1c1uyy8"
                    >
                      <h2
                        class="chakra-heading css-1ouru2w"
                      >
                        asdasd
                      </h2>
                      <p
                        class="chakra-text css-1vox59v"
                      >
                        asd@gmail.com
                      </p>
                    </div>
                  </div>
                </td>
                <td
                  class="css-1nygxpw"
                >
                  <div
                    class="chakra-stack css-qa66p0"
                  >
                    <h2
                      class="chakra-heading css-fcqnrt"
                    >
                      asd
                    </h2>
                    <p
                      class="chakra-text css-fcqnrt"
                    >
                      sad
                    </p>
                  </div>
                </td>
                <td
                  class="css-10ngfz9"
                >
                  <span
                    class="chakra-badge css-11k7pgp"
                  >
                    Offline
                  </span>
                </td>
                <td
                  class="css-svc9xl"
                >
                  <p
                    class="chakra-text css-1n7ogn7"
                  >
                    08/01/24
                  </p>
                </td>
                <td
                  class="css-lepxbj"
                >
                  <button
                    class="chakra-button css-yx91ij"
                    type="button"
                  >
                    Edit
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="chakra-stack css-1qzyrcn"
      >
        <div
          class="css-1ufrzn0"
        >
          <h2
            class="chakra-heading css-1fhgjcy"
          >
            Projects
          </h2>
        </div>
        <div
          class="chakra-table__container css-1c13jy5"
        >
          <table
            class="chakra-table css-0"
          >
            <thead
              class="css-0"
            >
              <tr
                class="css-0"
              >
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="COMPANIES"
                    >
                      COMPANIES
                    </p>
                  </div>
                </th>
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="BUDGET"
                    >
                      BUDGET
                    </p>
                  </div>
                </th>
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="STATUS"
                    >
                      STATUS
                    </p>
                  </div>
                </th>
                <th
                  class="css-xk75qt"
                >
                  <div
                    class="css-j2at52"
                  >
                    <p
                      class="chakra-text css-bfy219"
                      title="COMPLETION"
                    >
                      COMPLETION
                    </p>
                  </div>
                </th>
                <th
                  class="css-bsiblf"
                />
              </tr>
            </thead>
            <tbody
              class="css-0"
            >
              <tr
                class="css-0"
                data-testid="table-row"
              >
                <td
                  class="css-1ydvgvr"
                >
                  <div
                    class="css-fktblz"
                  >
                    <div
                      class="css-xax5g4"
                    >
                      <img
                        alt="Project "
                        class="chakra-image css-15sm3g6"
                        src="https://d38b044pevnwc9.cloudfront.net/cutout-nuxt/enhancer/3.jpg"
                      />
                    </div>
                    <div
                      class="chakra-stack css-1c1uyy8"
                    >
                      <h2
                        class="chakra-heading css-1ouru2w"
                      >
                        Project 
                      </h2>
                      <p
                        class="chakra-text css-1vox59v"
                      />
                    </div>
                  </div>
                </td>
                <td
                  class="css-os8l3e"
                >
                  <p
                    class="chakra-text css-ezs8j4"
                  >
                    $123,213
                  </p>
                </td>
                <td
                  class="css-10ngfz9"
                >
                  <span
                    class="chakra-badge css-sbi8wv"
                  >
                    Working
                  </span>
                </td>
                <td
                  class="css-1nygxpw"
                >
                  <div
                    class="chakra-stack css-1ey9w6j"
                  >
                    <p
                      class="chakra-text css-1dachdp"
                    >
                      12
                      %
                    </p>
                    <div
                      class="css-1cv4g7h"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="12"
                        class="css-1tm1yf1"
                        role="progressbar"
                        style="width: 12%;"
                      />
                    </div>
                  </div>
                </td>
                <td
                  class="css-lepxbj"
                >
                  <button
                    aria-controls="menu-list-:rb:"
                    aria-expanded="false"
                    aria-haspopup="menu"
                    class="chakra-button chakra-menu__menu-button css-14rg285"
                    id="menu-button-:rb:"
                    type="button"
                  >
                    <span
                      class="css-xl71ch"
                    >
                      <button
                        aria-label="This is the icon action"
                        class="chakra-button css-s1urw5"
                        data-testid="dot-icon"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          fill="none"
                          focusable="false"
                          height="4"
                          width="18"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M8 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0ZM1 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0ZM15 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0Z"
                            stroke="#A0AEC0"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </button>
                    </span>
                  </button>
                  <div
                    class="css-0"
                    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
                  >
                    <div
                      aria-orientation="vertical"
                      class="chakra-menu__menu-list css-6dyv79"
                      id="menu-list-:rb:"
                      role="menu"
                      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8);"
                      tabindex="-1"
                    >
                      <button
                        class="chakra-menu__menuitem css-bjyw8u"
                        data-index="0"
                        id="menu-list-:rb:-menuitem-:rc:"
                        role="menuitem"
                        tabindex="-1"
                        type="button"
                      >
                        <div
                          class="css-1tf9wat"
                        >
                          <svg
                            class="chakra-icon css-5au1dy"
                            data-testid="edit-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <g
                              fill="none"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-width="2"
                            >
                              <path
                                d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                              />
                              <path
                                d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                              />
                            </g>
                          </svg>
                          <svg
                            class="chakra-icon css-5au1dy"
                            data-testid="View-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <g
                              fill="currentColor"
                            >
                              <path
                                d="M23.432,10.524C20.787,7.614,16.4,4.538,12,4.6,7.6,4.537,3.213,7.615.568,10.524a2.211,2.211,0,0,0,0,2.948C3.182,16.351,7.507,19.4,11.839,19.4h.308c4.347,0,8.671-3.049,11.288-5.929A2.21,2.21,0,0,0,23.432,10.524ZM7.4,12A4.6,4.6,0,1,1,12,16.6,4.6,4.6,0,0,1,7.4,12Z"
                              />
                              <circle
                                cx="12"
                                cy="12"
                                r="2"
                              />
                            </g>
                          </svg>
                        </div>
                      </button>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
